shotover/src/sources/opensearch.rs

```diff
@@ -21,14 +23,38 @@ pub struct OpenSearchConfig {
 
 impl OpenSearchConfig {
     pub async fn get_source(
```

Comment:
This function is unused, why is it here?
Also check for other similar cases in the PR.

Reply:
Rust will give a warning if it detects unused functions, but this only works for private functions, declaring a function as `pub` disables this warning.